src/routes/v2/organizations.rs

```diff
@@ -0,0 +1,1038 @@
+use rust_decimal::Decimal;
+use serde::{Deserialize, Serialize};
+use sqlx::PgPool;
+use validator::Validate;
+
+pub fn config(cfg: &mut web::ServiceConfig) {
+    cfg.service(organizations_get).service(organization_create);
+    cfg.service(
+        web::scope("organization")
+            .service(organization_get)
+            .service(organizations_edit)
+            .service(organization_delete)
+            .service(organization_projects_get)
+            .service(organization_projects_add)
+            .service(organization_projects_remove)
+            .service(organization_icon_edit)
+            .service(delete_organization_icon)
+            .service(super::teams::team_members_get_organization),
+    );
+}
+
+#[derive(Deserialize, Validate)]
+pub struct NewOrganization {
+    #[validate(
+        length(min = 3, max = 64),
+        custom(function = "crate::util::validate::validate_name")
+    )]
+    pub name: String,
+    #[validate(length(min = 3, max = 256))]
+    pub description: String,
+    #[validate(
+        length(min = 3, max = 64),
+        regex = "crate::util::validate::RE_URL_SAFE"
+    )]
+    pub slug: String,
+    #[serde(default = "crate::models::teams::ProjectPermissions::default")]
+    pub default_project_permissions: ProjectPermissions,
+    #[validate(custom(function = "crate::util::validate::validate_urls"))]
+    /// Any desired links for the organization.
+    /// ie: "discord" -> "https://discord.gg/..."
+    #[serde(default)]
+    pub urls: HashMap<String, String>,
+}
+
+#[post("organization")]
+pub async fn organization_create(
+    req: HttpRequest,
+    new_organization: web::Json<NewOrganization>,
+    pool: web::Data<PgPool>,
+    redis: web::Data<deadpool_redis::Pool>,
+    session_queue: web::Data<AuthQueue>,
+) -> Result<HttpResponse, CreateError> {
+    let current_user = get_user_from_headers(
+        &req,
+        &**pool,
+        &redis,
+        &session_queue,
+        Some(&[Scopes::ORGANIZATION_CREATE]),
+    )
+    .await?
+    .1;
+
+    new_organization
+        .validate()
+        .map_err(|err| CreateError::ValidationError(validation_errors_to_string(err, None)))?;
+
+    let mut transaction = pool.begin().await?;
+
+    // Try slug
+    let slug_organization_id_option: Option<OrganizationId> =
+        serde_json::from_str(&format!("\"{}\"", new_organization.slug)).ok();
+
+    if let Some(slug_organization_id) = slug_organization_id_option {
+        let slug_organization_id: models::ids::OrganizationId = slug_organization_id;
+        let results = sqlx::query!(
+            "
+            SELECT EXISTS(SELECT 1 FROM organizations WHERE id=$1)
+            ",
+            slug_organization_id.0 as i64
+        )
+        .fetch_one(&mut *transaction)
+        .await
+        .map_err(|e| CreateError::DatabaseError(e.into()))?;
+
+        if results.exists.unwrap_or(false) {
+            return Err(CreateError::SlugCollision);
+        }
```

Comment:
you should just be able to use the organization get func here right>?

Reply:
Yeah, Im realizing I was copying behaviour in project_create that wasnt necessary (as it was created before redis was added). Maybe that should be changed too- what do you think? I can revisit project_create my testing PR